operator/iamkeycontroller/webhook.go

```diff
@@ -0,0 +1,76 @@
+package iamkeycontroller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	exoscalev1 "github.com/vshn/provider-exoscale/apis/exoscale/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// IAMKeyValidator validates admission requests.
+type IAMKeyValidator struct {
+	log logr.Logger
+}
+
+// ValidateCreate implements admission.CustomValidator.
+func (v *IAMKeyValidator) ValidateCreate(_ context.Context, obj runtime.Object) error {
+	iamKey := obj.(*exoscalev1.IAMKey)
+	v.log.V(1).Info("Validate create", "name", iamKey.Name)
+	if len(iamKey.Spec.ForProvider.Services.SOS.Buckets) == 0 {
+		return fmt.Errorf("an IAMKey named %q should have at least 1 allowed bucket",
+			iamKey.Name)
+	}
+	return nil
```

Comment:
As mentioned before, I would also check that the secret ref is given.

Reply:
Added the validation during `create` operation.